frontend/src/api/client.js

```diff
@@ -67,7 +69,18 @@ function createApiClient() {
 
     // Response interceptor
     instance.interceptors.response.use(
-        r => r,
+        (config) => {
+            const method = config.method?.toUpperCase();
+            if (["POST", "PUT", "PATCH", "DELETE"].includes(method)) {
+                const csrfToken = Cookies.get(CSRF_COOKIE_NAME);
+                if (csrfToken) {
+                    config.headers["X-CSRFToken"] = csrfToken;
+                }
+            }
+
+            config.withCredentials = true;
+            return config;
+        },
```

Comment:
The response interceptor is being used as a request interceptor, which can cause unexpected behavior. Use instance.interceptors.request.use for request modifications.
```
// Replace this block
instance.interceptors.request.use(
    (config) => {
        const method = config.method?.toUpperCase();
        if (["POST", "PUT", "PATCH", "DELETE"].includes(method)) {
            const csrfToken = Cookies.get(CSRF_COOKIE_NAME);
            if (csrfToken) {
                config.headers["X-CSRFToken"] = csrfToken;
            }
        }
        config.withCredentials = true;
        return config;
    },
    error => Promise.reject(error)
);
```